### ISSUE TITLE
Dogshell: Unicode error when piping/handling output containing international chars

### ISSUE THREAD

**Issue by samanthadrago** [DataDog/datadogpy#55]
I set up a monitor which contains an international character (ä).

When I run `dog monitor show_all` in the terminal, it outputs all monitors as expected.

But when I try to pipe the output or handle it at all, like `dog monitor show_all > dog.txt`, I get a Unicode error:

```
Traceback (most recent call last):
  File "/usr/local/bin/dog", line 9, in <module>
    load_entry_point('datadog==0.5.0', 'console_scripts', 'dog')()
  File "/usr/local/Cellar/python/2.7.9/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/datadog/dogshell/__init__.py", line 71, in main
    args.func(args)
  File "/usr/local/Cellar/python/2.7.9/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/datadog/dogshell/monitor.py", line 147, in _show_all
    (d["type"])]))
UnicodeEncodeError: 'ascii' codec can't encode character u'\xe4' in position 26: ordinal not in range(128)
```

The user who reported this used a workaround: on line 142 of datadog/dogshell/monitor.py, replace `(cls._escape(d["message"]))` with `(str(d["message"].encode('ascii','backslashreplace')))` but there may be a better way to handle this.

From: https://datadog.zendesk.com/agent/tickets/26802

**Comment by yannmh** [comment]
Moving to `0.11.0`.

**Comment by yannmh** [comment]
This exception is raised when a characters cannot be encoded with the standard output encoding. 

More specifically, Python 2.x defaults the standard output encoding to `ascii`, when redirecting to a file, i.e.
```
python2.7 -c "import sys; print sys.stdout.encoding" | cat
None
```
compared to
```
python2.7 -c "import sys; print sys.stdout.encoding"                                                                                
UTF-8
```

For this reason `dog monitor show_all` command works well in the terminal, but fails when redirected to a file.

I see different options to address this issue:

1. Escape non `ASCII` characters whatever the output encoding is set to.
2. Escape output encoding unsupported characters only.
3. Force/set the standard output encoding to `UTF-8`
4. Do nothing: standard output encoding can be set through the `PYTHONIOENCODING` environment variable. Potentially a warning message could invite users to set it.

Additionally I found this interesting reading about the subject: http://stackoverflow.com/questions/4545661/unicodedecodeerror-when-redirecting-to-file

**Comment by yannmh** [comment]
Closed with #268
